tests/kafkatest/tests/core/throttling_test.py

```diff
@@ -0,0 +1,173 @@
+        stop = time.time()
+        time_taken = stop - start
+        self.logger.debug("Transfer took %d second. Estimated time : %ds",
+                          time_taken,
+                          estimated_throttled_time)
+        assert time_taken >= estimated_throttled_time, \
+            ("Expected rebalance to take at least %ds, but it took %ds" % (
+                estimated_throttled_time,
+                time_taken))
+
+    @parametrize(bounce_brokers=False, new_consumer=True)
+    @parametrize(bounce_brokers=False, new_consumer=False)
+    def test_throttled_reassignment(self, bounce_brokers, new_consumer):
+        """Tests throttled partition reassignment. This is essentially similar
+        to the reassign_partitions_test, except that we throttle the reassignment
+        and verify that it takes a sensible amount of time given the throttle
+        and the amount of data being moved.
+
+        Since the correctness is time dependent, this test also simplifies the
+        cluster topology. We have 4 brokers, 1 topic with 4 partitions, and a
+        replication-factor of 1. The reassignment moves every partition. Hence
+        the data transfer in and out of each broker is fixed, and we can make
+        very accurate predicitons about whether throttling is working or not.
+        """
+
+        security_protocol = 'PLAINTEXT'
+        self.kafka.security_protocol = security_protocol
+        self.kafka.interbroker_security_protocol = security_protocol
+
+        producer_id = 'bulk_producer'
+        bulk_producer = ProducerPerformanceService(
+            context=self.test_context, num_nodes=1, kafka=self.kafka,
+            topic=self.topic, num_records=self.num_records,
+            record_size=self.record_size, throughput=-1, client_id=producer_id,
+            jmx_object_names=['kafka.producer:type=producer-metrics,client-id=%s' % producer_id],
+            jmx_attributes=['outgoing-byte-rate'])
+
+
+        self.producer = VerifiableProducer(context=self.test_context,
+                                           num_nodes=1,
+                                           kafka=self.kafka, topic=self.topic,
+                                           message_validator=is_int,
+                                           throughput=self.producer_throughput)
+
+        self.consumer = ConsoleConsumer(self.test_context,
+                                        self.num_consumers,
+                                        self.kafka,
+                                        self.topic,
+                                        new_consumer=new_consumer,
+                                        consumer_timeout_ms=60000,
+                                        message_validator=is_int,
+                                        from_beginning=False)
+
+        self.kafka.start()
+        bulk_producer.run()
+        self.run_produce_consume_validate(core_test_action=
```

Comment:
Yes. Part of the test is to make sure that we can produce and consume correctly during throttled reassignment, just like we check for unthrottled reassignment.